packages/legacy-client/src/client/actions/actions.ts

```diff
@@ -158,3 +182,15 @@ export type Actions<
         options?: Op,
       ) => WrappedActionReturnType<O, A, Op>;
 };
+
+export type BulkActions<O extends OntologyDefinition<any>> = {
+  [A in keyof O["actions"]]:
+    IsEmptyRecord<O["actions"][A]["parameters"]> extends true
+      ? <Op extends BulkActionExecutionOptions>(
+        options?: Op,
+      ) => WrappedBulkActionReturnType<O, A, Op>
```

Comment:
Right now I'm just following pattern of how we've handled other actions with no parameters, but the way its setup bulk actions don't really work for actions without parameters. This _could_ be modified so it accepts an array of `{}` potentially? And we just use those to tell us how many empty requests we need for the batch endpoint?

Reply:
Yeah, I think `{}[]` or `undefined[]` is better than nothing, otherwise you arent really calling bulk.

Reply:
Would `undefined[]` be preferable here? Because with `{}[]`won't that just let you pass in any sort of object https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcBCBXANgawIIDGMAlhAHYCiIwBGJ5A8mPWQM5wC8cA3gFBxwAthAAmwAPwAuOGQyCARsCgBufnCjAYGKJRHEYrKTLmKocXgF9eYglgCGGuADMMZIqTLOyACghhpmLiELFQ0dB5MLKwAlNIAbhDEIqq8Tj7c5gICAPTZcMRkMMBYWMSswGyorAAWEADu7AAGwmKNltG8vKCQsAjIqIH47uShtCyRHqwATJw8ahpaOhR6BkayCkrmVjb2ji5uLF5TADwACnCgRWQi7IPBHqPhjMyTUwB8vv5wp7FwCUmqNJTbx8LJCUTAaQAVlyMgg8CwwAA5nYsHAwA47IJ8k44IgIBg4AArDCseAQQT6dqdIHHO7DSjUMYRF7kVgfUFZFqQuAwvJkeFwREotEYqBYnF4gnE0nkykwaldcDQeBIFBweTYHDAQTMRCzTkFapKSHcCwAbQAuqorLwCGyEZoyVBEJJNbgdXqDUaTZJzdwAJIEOzXerVOxFOJKSQARgsABoeBZLRYgA

Reply:
But also feel like it could be weird synatx to do something like `client.ontology.bulkActions.noParamaction([undefined, undefined,...], {options})`?

Reply:
Oh wait could we use `Record<string,never>[]`?